internal/datacoord/job_manager.go

```diff
@@ -0,0 +1,294 @@
+package datacoord
+
+import (
+	"context"
+	"github.com/milvus-io/milvus-proto/go-api/v2/schemapb"
+	"sync"
+	"time"
+
+	"github.com/cockroachdb/errors"
+	"go.uber.org/zap"
+
+	"github.com/milvus-io/milvus/internal/datacoord/allocator"
+	"github.com/milvus-io/milvus/internal/proto/datapb"
+	"github.com/milvus-io/milvus/internal/proto/indexpb"
+	"github.com/milvus-io/milvus/pkg/log"
+	"github.com/milvus-io/milvus/pkg/util/merr"
+)
+
+type TaskManager interface {
```

Comment:
Better to rename to `StatsTaskManager` since it seems only manages the stats tasks. Then interface can be simplify to `SubmitTask`, `GetTaskState`, `DropTask`